tests/raftstore/test_transport.rs

```diff
@@ -0,0 +1,107 @@
+use super::cluster::{Cluster, Simulator};
+use super::node::new_node_cluster;
+use super::server::new_server_cluster;
+use std::collections::HashSet;
+
+fn split_leader_with_majority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    let mut s1 = HashSet::new();
+    let mut s2 = HashSet::new();
+    let mut vec: Vec<u64> = (1..count + 1).collect();
+    vec.swap(0, (leader - 1) as usize);
+    for i in 0..count / 2 + 1 {
+        s1.insert(vec[i as usize]);
+    }
+    for i in count / 2 + 1..count {
+        s2.insert(vec[i as usize]);
+    }
+    (s1, s2)
+}
+
+fn split_leader_with_minority(leader: u64, count: u64) -> (HashSet<u64>, HashSet<u64>) {
+    let mut s1 = HashSet::new();
+    let mut s2 = HashSet::new();
+    let mut vec: Vec<u64> = (1..count + 1).collect();
+    vec.swap(0, (leader - 1) as usize);
+    for i in 0..(count - 1) / 2 {
+        s1.insert(vec[i as usize]);
+    }
+    for i in (count - 1) / 2..count {
+        s2.insert(vec[i as usize]);
+    }
+    (s1, s2)
+}
+
+fn test_partition_stale_read<T: Simulator>(cluster: &mut Cluster<T>) {
+    let mut cluster = new_node_cluster(0, 5);
+    cluster.bootstrap_region().expect("");
+    cluster.start();
+
+    let (k, v) = (b"k1", b"v1");
+    cluster.put(k, v);
+    assert_eq!(cluster.get(k), Some(v.to_vec()));
+
+    let region_id = cluster.get_region_id(k);
+    let peer = cluster.leader_of_region(region_id).unwrap();
+    let leader = peer.get_store_id();
+
+    let (s1, s2) = split_leader_with_minority(leader, 5);
+    cluster.partition(&s1, &s2);
+
+    let v2 = b"v2";
+    cluster.put(k, v2);
```

Comment:
If it can put the new kv, it can either be that a new leader is elected or the partition filter is not working.